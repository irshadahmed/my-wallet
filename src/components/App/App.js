import React, { Component } from 'react';
import './App.scss';
import Wallet from '../../containers/Wallet/Wallet';
import { connect } from 'react-redux';
import BitcoinBalance from '../../containers/BitcoinBalance/BitcoinBalance';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return (
      <div className="app-container">
        <div className="app-wallet">
          <div className="ui card">
            <div className="content">
              <div className="header title">My Wallet App!</div>
            </div>
            <div className="extra content">
              <h3 className="balance">
                Wallet Balance: <span className="ui small header">{this.props.balance}</span>
              </h3>
            </div>
            <div className="content">
              <Wallet />
            </div>
            <div className="extra content">
              <BitcoinBalance />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ balance }) => ({ balance });

export default connect(mapStateToProps)(App);
