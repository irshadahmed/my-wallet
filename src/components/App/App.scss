.app-container {
  display: flex;
  justify-content: center;
  align-items: center;
  background-image: url('../../assets/background.jpg');
  background-size: 100%;
  height: 100%;

  .app-wallet {
    width: 80%;
    min-height: 200px;
    padding: 40px;
    .ui {
      &.card {
        width: 100%;
      }
    }
  }
}