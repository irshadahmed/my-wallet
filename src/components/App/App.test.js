import React from 'react';
import { shallow } from 'enzyme';
import  { shallowToJson } from 'enzyme-to-json';


import { App } from './App';

describe('App', () => {
  let wrapped;
  const props = {
    balance: 20,
  };

  beforeEach(() => {
    wrapped = shallow(<App {...props} />)
  });

  it('renders properly', () => {
    expect(shallowToJson(wrapped)).toMatchSnapshot();
  });

  it('displays the header text', () => {
    expect(wrapped.find('.title').text()).toEqual('My Wallet App!');
  });

  it('displays the balance from props', () => {
    expect(wrapped.find('.balance').text()).toEqual('Wallet Balance: 20');
  });

  it('contains a connected component called Wallet', () => {
    expect(wrapped.find('Connect(Wallet)').exists()).toEqual(true);
  });

  it('contains a connected component called BitcoinBalance', () => {
    expect(wrapped.find('Connect(BitcoinBalance)').exists()).toEqual(true);
  });

});