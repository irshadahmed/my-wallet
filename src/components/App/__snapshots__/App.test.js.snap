// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div
  className="app-container"
>
  <div
    className="app-wallet"
  >
    <div
      className="ui card"
    >
      <div
        className="content"
      >
        <div
          className="header title"
        >
          My Wallet App!
        </div>
      </div>
      <div
        className="extra content"
      >
        <h3
          className="balance"
        >
          Wallet Balance: 
          <span
            className="ui small header"
          >
            20
          </span>
        </h3>
      </div>
      <div
        className="content"
      >
        <Connect(Wallet) />
      </div>
      <div
        className="extra content"
      >
        <Connect(BitcoinBalance) />
      </div>
    </div>
  </div>
</div>
`;
